src/patterns/flyweight/avatar-example/flyweight-factory/flyweight-factory.impl.ts

```diff
@@ -15,12 +15,12 @@ export class FlyweightFactory<K, S, F> implements IFlyweightFactory<K, S, F> {
     const key = this.generateKey(sharedState);
 
     if (!this.flyweights.has(key)) {
-      const newFlyweight = new Flyweight<F>(this.generateIntrinsicState(sharedState)).sharedState as F;
+      const newFlyweight = new Flyweight<F>(this.generateIntrinsicState(sharedState)).sharedState;
 
       this.flyweights.set(key, newFlyweight);
     }
 
-    return this.flyweights.get(key) as F;
+    return this.flyweights.get(key)!;
```

Comment:
the same reason: 
![image](https://github.com/user-attachments/assets/e6a19b5f-52a7-49ca-94ae-2396366d86cf)